utils/rjson.cc

```diff
@@ -251,7 +251,8 @@ future<> print(const rjson::value& value, seastar::output_stream<char>& os, size
                 // it might be a good idea to instead send our buffer as a packet directly. If so, the
                 // buffer size should probably increase (at least after first send()).
                 _f = _f.then([this, buf = std::move(_buf), &os = _os, try_reuse]() mutable -> future<> {
-                    return os.write(buf.get(), buf.size()).then([this, buf = std::move(buf), try_reuse]() mutable {
+                    co_return co_await os.write(buf.get(), buf.size()).then([this, buf = std::move(buf), try_reuse]() mutable {
+
```

Comment:
https://github.com/scylladb/seastar/blob/master/doc/tutorial.md#lambda-coroutines

Mixing coroutines and continuations is a recipe for confusion. Better to coroutinize everything.

Reply:
I see the caller isn't a coroutine. Better to use non-coroutine yields then (seastar::maybe_yield()).

Reply:
By the way, does co_await automatically add preemption? If not, what in the above change adds preemption?
How did you test this works?

Reply:
> By the way, does co_await automatically add preemption? 

Yes.